wsgi/protocolhttp.cpp

```diff
@@ -326,6 +326,17 @@ void ProtocolHttp::parseHeader(const char *ptr, const char *end, Socket *sock) c
     } else if (!protoRequest->headerHost && key == QLatin1String("HOST")) {
         protoRequest->serverAddress = value;
         protoRequest->headerHost = true;
+    } else if (usingFrontendProxy && !protoRequest->X_Forwarded_For && key == QLatin1String("X_FORWARDED_FOR")) {
+        protoRequest->remoteAddress = QHostAddress(value); // configure your reverse-proxy to list only one IP address
+        protoRequest->remotePort = 0; // unknown
+        protoRequest->X_Forwarded_For = true;
+    } else if (usingFrontendProxy && !protoRequest->X_Forwarded_Host && key == QLatin1String("X_FORWARDED_HOST")) {
+        protoRequest->serverAddress = value;
+        protoRequest->X_Forwarded_Host = true;
```

Comment:
This seems a useless variable X_Forwarded_Host as it's the same for headerHost, I personally set NGINX to set the 'Host' header for me on websockets.

Reply:
I used [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) as a reference. I personally also have Nginx rewrite the Host. So I'm fine with the removal of this option.

Reply:
btw: X-Real-Ip is the same as X_Forwarded_For. I can include it as a fallback, if you like

Reply:
> I used [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) as a reference. I personally also have Nginx rewrite the Host. So I'm fine with the removal of this option.

No I was not saying to remove the option, just the bool variable that you can just use headerHost.

Reply:
ok

Reply:
> btw: X-Real-Ip is the same as X_Forwarded_For. I can include it as a fallback, if you like

hmm if so then it's probably interesting to set the cutelyst remoteAddress variable to the IP on this header

Reply:
once again X_Forwarded_Host: if I use headerHost, I guess the header X-Forwarded-Host is never used, because a http request has to have a 'Host' header, right? So, you are required to configure nginx to rewrite it, otherwise an additional X-Forwarded-Host is never used. Is this your intention?

Reply:
ah ok, you are right, nvm